sql/src/main/java/io/druid/sql/calcite/schema/SystemSchema.java

```diff
@@ -0,0 +1,435 @@
+      final List<DataSegment> matadataSegments = druidDataSourceList
+          .stream()
+          .flatMap(t -> t.getSegments().stream())
+          .collect(Collectors.toList());
+      final Map<String, DataSegment> publishedSegments = matadataSegments
+          .stream()
+          .collect(Collectors.toMap(
+              DataSegment::getIdentifier,
+              Function.identity()
+          ));
+      final Set<String> publishedSegmentIds = matadataSegments
+          .stream()
+          .map(DataSegment::getIdentifier)
+          .collect(Collectors.toSet());
+
+      final Map<String, QueryableDruidServer> serverViewClients = serverView.getClients();
+      for (QueryableDruidServer queryableDruidServer : serverViewClients.values()) {
+        final Object[] row = new Object[SEGMENTS_TABLE_SIZE];
+        final DruidServer druidServer = queryableDruidServer.getServer();
+        final ServerType type = druidServer.getType();
+        final boolean isSegmentReplicatable = druidServer.segmentReplicatable();
+        final Map<String, DataSegment> segments = new HashMap<>(druidServer.getSegments());
+        segments.putAll(publishedSegments);  // add published segments map
+        for (Map.Entry<String, DataSegment> segmentEntry : segments.entrySet()) {
+          String segmentId = segmentEntry.getKey();
+          DataSegment segment = segmentEntry.getValue();
+          row[0] = segment.getIdentifier();
+          row[1] = segment.getDataSource();
+          row[2] = segment.getInterval().getStart();
+          row[3] = segment.getInterval().getEnd();
+          boolean is_available = false;
+          boolean is_published = false;
+          boolean is_realtime = false;
+          if (publishedSegmentIds.contains(segmentId)) {
+            is_published = true;
+          }
+          if (type.toString().equals(ServerType.HISTORICAL.toString())) {
+            is_available = true;
+            is_published = true;
+          } else if (type.toString().equals(ServerType.REALTIME.toString())) {
+            is_available = true;
+            is_realtime = true;
+          } else if (type.toString().equals(ServerType.INDEXER_EXECUTOR.toString())) {
+            is_available = true;
+            is_published = true;
+            is_realtime = true;
+          }
+          if (!isSegmentReplicatable) {
+            is_realtime = true;
+          }
+          row[4] = is_published;
+          row[5] = is_available;
+          row[6] = is_realtime;
+          try {
+            String payload = jsonMapper.writeValueAsString(segment);
+            row[7] = payload;
+          }
+          catch (JsonProcessingException e) {
+            log.error(e, "Error parsing segment payload ");
+          }
+          rows.add(row);
+        }
+      }
+      return Linq4j.asEnumerable(rows);
+    }
+
+    private List<ImmutableDruidDataSource> getMetadataSegments(
+        DruidLeaderClient druidLeaderClient,
```

Comment:
Let's name this appropriately as to what it is (coordinator client).

Reply:
renamed